src/Nethermind/Nethermind.Specs/ChainSpecStyle/ChainSpecBasedSpecProvider.cs

```diff
@@ -276,6 +278,15 @@ public IReleaseSpec GetSpec(ForkActivation activation)
                 {
                     if (_logger.IsWarn) _logger.Warn($"Chainspec file is misconfigured! Timestamp transition is configured to happen before the last block transition.");
                 }
+
+                if (_firstTimestampActivation.Value.Timestamp <= activation.Timestamp)
```

Comment:
would be a bit simpler if in the `if` you only choose which array to look:
something like:
```suggestion
                (ForkActivation Activation, ReleaseSpec Spec)[] consideredTimestamps =   _firstTimestampActivation.Value.Timestamp <= activation.Timestamp ?_timestampOnlyTransitions : _transitions;
```